rs-react-app/src/components/header/header.tsx

```diff
@@ -0,0 +1,23 @@
+import { Component } from 'react';
+import Input from './search/Input.tsx';
+import Button from './search/Button.tsx';
+import './header.css';
+import type { HeaderProps } from '../../interfaces/interface';
```

Comment:
It is more comfy to store types and interfaces for component's props and/or state here in the corresponding `.tsx` file